internal/proxy/recorder.go

```diff
@@ -60,7 +62,16 @@ func StartRecorder(upstream string, dir string, options RecorderOptions) (chan H
 			exchange := <-recordC
 
 			var responseFilePrefix string
-			requestHash := getRequestHash(exchange.Request)
+			var requestHash string
+			
+			if options.Soap11Mode {
+				// In SOAP mode, use SOAPAction + path for uniqueness
+				soapAction := extractSoapAction(exchange.Request)
+				requestHash = getRequestHash(exchange.Request) + "_" + soapAction
```

Comment:
It would be better to extend the `getRequestHash` function to be a variadic function and take one or more strings to include in the hash computation. In SOAP mode, we can pass the SOAPAction, and nothing in the normal case.